lib/pages/channel/channel_files/channel_files_widget.dart

```diff
@@ -0,0 +1,128 @@
+import 'package:twake/widgets/common/file_channel_tile.dart';
+
+class ChannelFilesWidget extends StatefulWidget {
+  const ChannelFilesWidget({Key? key}) : super(key: key);
+
+  @override
+  _ChannelFilesWidgetState createState() => _ChannelFilesWidgetState();
+}
+
+class _ChannelFilesWidgetState extends State<ChannelFilesWidget> {
+
+  late final Channel? _currentChannel;
+  final channelFileCubit = Get.find<ChannelFileCubit>();
+
+  @override
+  void initState() {
+    super.initState();
+    _currentChannel = Get.arguments;
+    WidgetsBinding.instance?.addPostFrameCallback((_) {
+      if(_currentChannel != null) {
+        channelFileCubit.loadFilesInChannel(_currentChannel!.id);
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        child: Column(
+          children: [
+            _buildHeader(),
+            _buildListFiles(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  _buildHeader() {
+    return Container(
+      color: Theme.of(context).colorScheme.secondaryContainer,
+      height: 56,
+      child: Stack(
+        children: [
+          Align(
+            alignment: Alignment.centerLeft,
+            child: IconButton(
+                onPressed: () => popBack(),
+                icon: Icon(
+                  Icons.arrow_back_ios,
+                  color: Theme.of(context).colorScheme.surface,
+                )),
+          ),
+          Align(
+            alignment: Alignment.center,
+            child: Container(
+              alignment: Alignment.center,
+              width: 170,
+              child: AutoSizeText(
+                AppLocalizations.of(context)!.files,
+                maxLines: 1,
+                maxFontSize: 17,
+                minFontSize: 12,
+                style: Theme.of(context)
+                    .textTheme
+                    .headline1!
+                    .copyWith(
+                    fontWeight: FontWeight.w600, fontSize: 17),
+              ),
+            ),
+          )
+        ],
+      ),
+    );
+  }
+
+  _buildListFiles() {
+    return BlocBuilder<ChannelFileCubit, ChannelFileState>(
+      bloc: channelFileCubit,
+      builder: (context, state) {
+        if(state.channelFileStatus == ChannelFileStatus.finished) {
+          final channels = state.listFiles;
+          return Expanded(
+            child: Container(
+              padding: const EdgeInsets.symmetric(vertical: 16.0, horizontal: 12.0),
+              child: ListView.separated(
+                padding: const EdgeInsets.all(6),
+                shrinkWrap: true,
+                itemCount: channels.length,
+                separatorBuilder: (BuildContext context, int index) => SizedBox(height: 12.0),
+                itemBuilder: (context, index) {
+                  return _buildChannelFileItem(channels[index]);
+                },
+              ),
+            ),
+          );
+        }
+        return Container(
```

Comment:
What if we don't have files in the channel - show the CircularProgressIndicator forever, this needs another solution for that case